spring-boot-autoconfigure/src/main/java/org/axonframework/springboot/autoconfig/JpaEventStoreAutoConfiguration.java

```diff
@@ -0,0 +1,45 @@
+package org.axonframework.springboot.autoconfig;
+
+import org.axonframework.common.jdbc.PersistenceExceptionResolver;
+import org.axonframework.common.jpa.EntityManagerProvider;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventsourcing.eventstore.EventStorageEngine;
+import org.axonframework.eventsourcing.eventstore.EventStore;
+import org.axonframework.eventsourcing.eventstore.jpa.JpaEventStorageEngine;
+import org.axonframework.serialization.Serializer;
+import org.axonframework.spring.config.AxonConfiguration;
+import org.axonframework.springboot.util.RegisterDefaultEntities;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+import javax.persistence.EntityManagerFactory;
+
+@ConditionalOnBean(EntityManagerFactory.class)
+@ConditionalOnMissingBean({EventStorageEngine.class, EventStore.class})
+@RegisterDefaultEntities(packages = {
+        "org.axonframework.eventsourcing.eventstore.jpa"
+})
+@Configuration
+public class JpaEventStoreAutoConfiguration {
```

Comment:
Doesn't this split warrant a split of the test cases too?

Reply:
I've added some tests for both `JpaAutoConfiguration `and `JpaEventStoreAutoConfiguration`